pkg/resource/service.go

```diff
@@ -373,10 +411,27 @@ func (r *ServiceResource) Create(ctx context.Context, req resource.CreateRequest
 	}
 
 	if service.Tier == api.TierProduction {
-		minTotalMemoryGb := int(plan.MinTotalMemoryGb.ValueInt64())
-		service.MinTotalMemoryGb = &minTotalMemoryGb
-		maxTotalMemoryGb := int(plan.MaxTotalMemoryGb.ValueInt64())
-		service.MaxTotalMemoryGb = &maxTotalMemoryGb
+		var minReplicaMemoryGb, maxReplicaMemoryGb int
+		if !plan.MinReplicaMemoryGb.IsNull() {
+			minReplicaMemoryGb = int(plan.MinReplicaMemoryGb.ValueInt64())
+		} else {
+			// Due to a bug on the API, we always assumed the MinTotalMemoryGb value was always related to 3 replicas.
+			// Now we use a per-replica API to set the min total memory so we need to divide by 3 to get the same
+			// behaviour as before.
+			minReplicaMemoryGb = int(plan.MinTotalMemoryGb.ValueInt64() / 3)
```

Comment:
Maybe worth using num_replicas instead of hardcoded `3`?

Reply:
discussed offline. hardcoded 3 is what we need to solve the issue